designer/server/src/main/scala/pl/touk/nussknacker/ui/validation/ProcessValidation.scala

```diff
@@ -103,8 +106,8 @@ class ProcessValidation(
   }
 
   def uiValidation(displayable: DisplayableProcess): ValidationResult = {
-    validateIds(displayable)
-      .add(validateEmptyId(displayable))
+    validateScenarioId(displayable)
```

Comment:
Why do you do these validation here if you also do them on canonicalprocess (in ProcessCompiler)?

Reply:
Previously, FE validated empty node/scenario id independently so if you had errors that prevented converting DisplayableProcess to CanonicalProcess, the empty id errors still appeared. Now, if we do the validation only on the CanonicalProcess level, we lose this validation when we cant canonize the graph.

This generates some problems with the need to deduplicate these errors, but it's the only solution I've found for this validation to work as previously. This may not be worth because the gain is minimal for the complexity.

Reply:
Uh I see. I have no strong opinion on that. I see two options:
1. We can add a comment why it is duplicated
2. We can remove this step, lost this small benefit and revert it if we find out that it was crucial for someone

Reply:
I've added a comment at https://github.com/TouK/nussknacker/pull/4745/files#diff-d4b08503fe91601c6e3b096ed946441b1a93faa59a8dc5467b6e64b2cbf47991R86

Reply:
It is nice, but I would also add a comment before `uiValidation` method that we duplicate some validation invocation because `uiValidation` is the first step that stops further validations and in case when we have some error we don't want to loose these validation errors.

Reply:
Good point, done.